classlib/src/main/java/org/teavm/classlib/java/util/TList.java

```diff
@@ -159,4 +208,78 @@ static <E> TList<E> of(E... elements) {
     static <E> TList<E> copyOf(TCollection<? extends E> collection) {
         return new TTemplateCollections.ImmutableArrayList<>(collection);
     }
+
+    class ReversedList<E> extends TAbstractList<E> {
```

Comment:
I'd brought it to top level. And made it package-private

Reply:
Implementing `iterator` would also be quite useful here

Reply:
It uses implementation of iterator from AbstractList which is simple and performant.

Reply:
It's not. Underlying list does not necessarily implement `get` in the most efficient ways. Sometimes it's much faster to iterate over list than getting elements with `get` method.